sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -1059,6 +1059,24 @@ ssize_t coap_opt_add_proxy_uri(coap_pkt_t *pkt, const char *uri)
     return _add_opt_pkt(pkt, COAP_OPT_PROXY_URI, (uint8_t *)uri, strlen(uri));
 }
 
+ssize_t coap_opt_replace_etag(coap_pkt_t *pkt, const uint8_t *etag, size_t len)
+{
+    coap_optpos_t optpos = { 0 };
+    uint8_t *val = NULL;
+    ssize_t res = coap_opt_get_next(pkt, &optpos, &val, true);
+    while (optpos.opt_num != COAP_OPT_ETAG) {
```

Comment:
Why not use `coap_find_option(pkt, COAP_OPT_ETAG)`?

Reply:
Because, although it doesn't matter for creating packets, I don't like how `coap_find_option` modifies the `coap_pkt_t`. Always feels like a future footgun to me

Reply:
uh yea it appears to be intended only for the receiving side, when the CoAP packet already has been parsed.

Reply:
We should really make that distinction in the type, even if it's just through a type alias.